### PR TITLE
docs: Add example for Jenkins pipeline (accessing self-hosted GitLab)

### PR DESCRIPTION
**Pull request by florianklumb**
As promised in https://github.com/containerbase/buildpack/discussions/370

### CODE REVIEW

**Comment by florianklumb** [comment]
> Please add a link from root readme to this file.

Done

File: docs/jenkins.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        LOG_LEVEL = 'debug'
+        GIT_AUTHOR_NAME = 'Renovate Bot'
+        GIT_AUTHOR_EMAIL = 'bot@example.com'
+        GIT_COMMITTER_NAME = 'Renovate Bot'
+        GIT_COMMITTER_EMAIL = 'bot@example.com'
+        RENOVATE_PLATFORM = 'gitlab'
+        RENOVATE_ENDPOINT = 'https://git.example.com/api/v4/'
+        RENOVATE_REPOSITORIES = 'user1/repo1, user2/repo2'
+        RENOVATE_ONBOARDING_CONFIG = '{ "extends":["config:base"] }'
+        RENOVATE_BINARY_SOURCE = 'install'
```
Comment (viceice) [inline docs/jenkins.md:50]: you don't need to map docker socket for this mode.

you should add two samples
1. docker mode with slim image
2. install mode with full image

File: docs/jenkins.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        }
+    }
+}
+
+```
+
+### Renovate specific config explained
+
+- `args '... --group-add 0'`: give Docker container user root rights
+- `RENOVATE_BINARY_SOURCE = 'install'`: allows to use npm and node versions explicitely set in the `engines` section of your `package.json` for renovation, especially to keep RenovateBot from downgrading package-lock.json to v1 (see also [Renovate downgraded npm lockfile version from v2 to v1](https://github.com/renovatebot/renovate/discussions/14409))
```
Comment (viceice) [inline docs/jenkins.md:97]: this is wrong, if you like to use docker side car, this should be removed and you should use the slim image, which sets this to  `docker`

**Review by viceice — CHANGES_REQUESTED** [review]

File: docs/jenkins.md (hunk 1 above)
Comment (florianklumb) [reply docs/jenkins.md:50]: @viceice , I'm sorry. I'm not an expert in these things and have only limited time. This is what I did, what I can vouch for in terms of that it works and what I can offer you at this point. Please accept this as a base which can then be extended by someone else.

File: docs/jenkins.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+- `RENOVATE_EXTRA_FLAGS`: pass additional commandline args (**optional**)
+
+### Jenkinsfile
+```yml
+#!groovy
+
+pipeline {
+    agent {
+        docker {
+            image 'renovate/renovate:32.32.5'
```
Comment (viceice) [inline docs/jenkins.md:30]:
```suggestion
            image 'renovate/renovate:32.32.5-slim'
```

use slim image with docker

File: docs/jenkins.md (hunk 1 above)
Comment (viceice) [reply docs/jenkins.md:50]:
```suggestion
```

Simply remove, because you like to use the docker sidecar

File: docs/jenkins.md (hunk 2 above)
Comment (viceice) [reply docs/jenkins.md:97]:
```suggestion
```

**Comment by florianklumb** [comment]
So?

**Review by viceice — COMMENTED** [review]
Did you tested this setup?

**Comment by florianklumb** [comment]
> Did you tested this setup?

Seem to work, yes.

**Comment by viceice** [comment]
yes

File: docs/jenkins.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            archiveArtifacts allowEmptyArchive: true, artifacts: 'renovate.log'
+        }
+    }
+}
+
+```
+
+### Renovate specific config explained
+
+- `args '... --group-add 0'`: give Docker container user root rights
```
Comment (viceice) [inline docs/jenkins.md:94]:
```suggestion
- `args '... --group-add 0'`: give Docker container user root group rights to some required files and folders
```

**Review by viceice — APPROVED** [review]

**Pull request merged**